drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,538 @@
+			LOG_ERR("unable to set control register");
+			return result;
+		}
+	}
+
+	data->previous_state = data->state;
+
+	return 0;
+}
+
+static int tle9104_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct tle9104_data *data = dev->data;
+	int result;
+
+	/* cannot execute a bus operation in an ISR context */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	if (pin > 7) {
+		LOG_ERR("invalid pin nummber %i", pin);
+		return -EINVAL;
+	}
+
+	if ((flags & GPIO_INPUT) != 0) {
+		LOG_ERR("cannot configure pin as input");
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_OUTPUT) == 0) {
+		LOG_ERR("pin must be configured as an output");
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_SINGLE_ENDED) == 0) {
+		LOG_ERR("pin must be configured as single ended");
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_LINE_OPEN_DRAIN) == 0) {
+		LOG_ERR("pin must be configured as open drain");
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_UP) != 0) {
+		LOG_ERR("pin cannot have a pull up configured");
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_DOWN) != 0) {
+		LOG_ERR("pin cannot have a pull down configured");
+		return -ENOTSUP;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+		WRITE_BIT(data->state, pin, 0);
+	} else if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+		WRITE_BIT(data->state, pin, 1);
+	}
+
+	WRITE_BIT(data->configured, pin, 1);
+	result = tle9104_write_state(dev);
+	k_mutex_unlock(&data->lock);
+
+	return result;
+}
+
+static int tle9104_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	LOG_ERR("input pins are not available");
+	return -ENOTSUP;
+}
+
+static int tle9104_port_set_masked_raw(const struct device *dev, uint32_t mask, uint32_t value)
+{
+	struct tle9104_data *data = dev->data;
+	int result;
+
+	/* cannot execute a bus operation in an ISR context */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+	data->state = (data->state & ~mask) | (mask & value);
+	result = tle9104_write_state(dev);
+	k_mutex_unlock(&data->lock);
+
+	return result;
+}
+
+static int tle9104_port_set_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return tle9104_port_set_masked_raw(dev, mask, mask);
+}
+
+static int tle9104_port_clear_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return tle9104_port_set_masked_raw(dev, mask, 0);
+}
+
+static int tle9104_port_toggle_bits(const struct device *dev, uint32_t mask)
+{
+	struct tle9104_data *data = dev->data;
+	int result;
+
+	/* cannot execute a bus operation in an ISR context */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+	data->state ^= mask;
+	result = tle9104_write_state(dev);
+	k_mutex_unlock(&data->lock);
+
+	return result;
+}
+
+static int tle9104_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					   enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	return -ENOTSUP;
+}
```

Comment:
Please add `ARG_UNUSED()` macro calls for all unused arguments.

Reply:
done